### PR TITLE
A trip down the rabbit hole to enable colors from sbt

### PR DESCRIPTION
**Pull request by szeiger**
I set out (and succeeded) to enable color output when running from sbt. Along the way I refactored the outer layers of partest (basically everything above `SuiteRunner`) to remove global mutable state and make it more manageable.

### CODE REVIEW

**Comment by szeiger** [comment]
Builds failed because of binary compatibility checks, as expected. This is obviously not binary compatible but shouldn't touch the parts of partest needed by partest-extras.

**Comment by SethTisue** [comment]
can you suggest a reviewer? I don't know this codebase.

**Comment by szeiger** [comment]
@adriaanm perhaps? He must have touched at least parts of this code while adding sbt support

**Comment by som-snytt** [comment]
How I miss the coleurs.

**Comment by adriaanm** [comment]
Let's merge this and bootstrap a new version of partest after M5. 🌈

**Comment by szeiger** [comment]
Rebased. Rebasing on top of #63 was a PITA. @som-snytt, can you take a look to make sure I didn't break anything?

File: src/main/scala/scala/tools/partest/sbt/SBTRunner.scala [hunk 1]
```diff
@@ -6,12 +6,15 @@
 **                          |/                                          **
 \*                                                                      */
 
-package scala.tools.partest
-package nest
+package scala.tools.partest.sbt
```
Comment (lrytz) [inline src/main/scala/scala/tools/partest/sbt/SBTRunner.scala]: who is using this class? can we easily just move it and update its users simultaneously?

**Comment by lrytz** [comment]
LGTM otherwise. did you do some testing locally? (i didn't)

File: src/main/scala/scala/tools/partest/sbt/SBTRunner.scala (hunk 1 above)
Comment (szeiger) [inline src/main/scala/scala/tools/partest/sbt/SBTRunner.scala]: It's used by `PartestTask` in the same package. This used to be in `partest-interface` before we moved it into partest itself.

**Comment by szeiger** [comment]
@lrytz I'm pretty sure tested it locally when I wrote it. I don't remember exactly what I tested because it's been a while.

**Comment by som-snytt** [comment]
@szeiger Sorry about the conflict. It would have been easier for me to rebase.

I took a quick look, but I won't have more time until end of day. I'll run it later, one way or another.

**Pull request merged**